React/Views/ScrollView/RCTScrollView.m

```diff
@@ -817,12 +821,36 @@ - (void)scrollViewDidEndZooming:(UIScrollView *)scrollView withView:(UIView *)vi
   RCT_FORWARD_SCROLL_EVENT(scrollViewDidEndZooming : scrollView withView : view atScale : scale);
 }
 
+- (void)didMoveToWindow
+{
+  [super didMoveToWindow];
+  if (self.window == nil) {
+    if(_isScrollBeginFired == YES && _isScrollEndFired == NO) {
+      // Fire the end deceleration event
+      RCT_SEND_SCROLL_EVENT(onMomentumScrollEnd, nil);
+      RCT_FORWARD_SCROLL_EVENT(scrollViewDidEndDecelerating : _scrollView);
+      _isScrollEndFired = YES;
+      _isScrollBeginFired = NO;
+    }
+  }
+}
+
+- (void)scrollViewWillBeginDecelerating:(UIScrollView *)scrollView
+{
+  _isScrollBeginFired = YES;
```

Comment:
if the view is starting decelerating, it means it is stopping. The ScrollBegin has been fired long ago. is there a way to set `_isScrollBeginFired` earlier, when the scroll actually starts?